CRM/Contribute/Form/ContributionBase.php

```diff
@@ -332,9 +339,8 @@ public function preProcess() {
     $this->_bltID = $this->get('bltID');
     $this->_paymentProcessor = $this->get('paymentProcessor');
 
-    // This get will ensure it is set for later. Once we are sure all places
-    // access via the get & not directly, it can go.
-    $this->getPriceSetID();
+    $this->order = new CRM_Financial_BAO_Order();
+    $this->order->setPriceSetID($this->getPriceSetID());
```

Comment:
I added in the `order` object here - we are only using it to get the cached price set metadata but it has a lot of helpers that can, in future fixes, remove a lot of the code in these classes. In general I want things that can be calculated to not be passed around / only to be set as properties within functions that expose how they would be calculated - so using the cached price set metadata is the direction I want to go. The order class functions will replace `init` whic is where `_priceSet` is otherwise retrieved from